README.md

```diff
@@ -2,272 +2,396 @@
-// LOG: legume: Peanut
-// ERROR: Uncaught ReferenceError: vegetable is not defined
+function addStrike() {
+  const totalStrikes = strikes + 1;
+  console.log(player + " has " + totalStrikes + " strike(s)!");
+
+  return totalStrikes;
+}
 ```
 
-Okay, we have an idea of what the scope chain is, but how does it actually work
-under the hood?
+When declaring `totalStrikes`, another variable is utilized in its definition,
+`strikes`. JavaScript will see this variable, and first try to look for it
+within the current scope of `addStrike()`. When it doesn't find it anywhere in
+`addStrike()`, it will go look for it in the closest outer scope in the chain.
+In this case, the `isOut()` function scope. Once it discovers that a `strikes`
+variable does exist in that scope, it will use its value and stop traversing the
+chain.
 
-## The JavaScript engine and identifier resolution
+This is the general process of how JavaScript uses the scope chain to resolve
+variable lookup.
 
-### Identifiers
+A similar process happens in the next line of code. In the `console.log()`, the
+`player` variable is used, but it does not exist within `addStrike()`. When
+JavaScript looks for it in the closest outer scope, it cannot find it in
+`isOut()` either. So, it goes one step further out to the global scope, where it
+successfully finds `player`'s value.
 
-As a brief refresher, when we declare a variable or a function, we provide a
-name that allows us to refer back to it:
+**When looking up variables, JavaScript will always look at its current scope
+first. Then, if unfound, it will continue moving up the scope chain until the
+the variable's value is found.**
 
-```js
-const myVar = "myVar refers to the variable that contains this string";
-// => undefined
+Next, let's take a look at the `if` statement block.
 
-function myFunc() {
-  return "myFunc refers to this function that returns this string";
+```js
+if (strikes >= availableStrikes) {
+  atBat = false;
+  const message = player + " is out!";
+  console.log(message);
 }
-// => undefined
 ```
 
-We call those names _identifiers_ because they allow us to **identify** the
-variable or function we're referring to.
+For the conditional, it uses variables from the `isOut()` scope that it has
+access to. It also uses a variable, `player`, from the global scope when
+declaring `message`. In all these cases, the process is as we just discussed. It
+will keep going up the scope chain until it finds the variables needed.
 
-### The JavaScript engine
+The new point we want to pay attention to here is the value reassignment of the
+`atBat` variable.
 
-When our JavaScript code is run in the browser, the JavaScript engine actually
-makes two separate passes over our code:
+As we previously learned, when _declaring_ a variable with no keyword, it will
+become a globally scoped variable, regardless of where it's defined. In this
+case, we intuitively know, from what we've learned about JavaScript so far, that
```

Comment:
I recommend modifying the second sentence to just: "However, that isn't what's happening here."